### ISSUE TITLE
Temp stash those unstaged files when running cactus

### ISSUE THREAD

**Issue by lemorage** [bwireman/cactus#10]
I am using Cactus for git pre-commit hooks, but I’ve encountered an issue where it keeps failing, even though the files causing the errors are not in my staging area. To pass the Cactus checks, I’m forced to fix the errors in those files (or stash them offhand), even though they aren’t part of the commit I’m making.

I’m wondering if it’s possible to configure Cactus to temporarily stash these files while running the pre-commit checks. Once the commit is complete, the stashed files can then be restored. It would be nice if we could use Cactus to automate this process.

Example as follows:
```shell
   Compiled in 0.02s
    Running cactus.main
Initializing hook: 'pre-commit'
Running 'gleam format --check'
Running 'gleam run -m go_over -- --outdated'
'pre-commit' hook failed. Reason: Action Failed Error:
error: These files have not been formatted

  - ./src/ganz.gleam       # <--- but this file is not staged
```

**Comment by bwireman** [comment]
That's a great suggest, I'll do some digging. For the time being you could consider 
1. adding steps to stash and unstash files to your pre-commit hooks
2. move to pre-push hooks, assuming your workflows tend to work out that way

**Comment by bwireman** [comment]
Included in [v1.3.0](https://hex.pm/packages/cactus)
